From: Maytham Alsudany <maytha8thedev@gmail.com>
Date: Wed, 21 May 2025 10:45:21 +0200
Subject: skip publicsuffix tests

Bug-Debian: https://bugs.debian.org/1089192
Forwarded: not-needed
Last-Update: 2025-05-03

These tests are hardcoded by upstream to check against parts of the test data,
but break when the test data is regenerated from the latest publicsuffix data,
which is constantly changing.

This patch skips TestPublicSuffix, TestSlowPublicSuffix, TestNumICANNRules,
ExamplePublicSuffix_manager, and TestICANN, which are all affected by this
problem and resulted in test failures.
---
 publicsuffix/example_test.go | 4 +++-
 publicsuffix/list_test.go    | 4 ++++
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/publicsuffix/example_test.go b/publicsuffix/example_test.go
index c051dac..bc5f733 100644
--- a/publicsuffix/example_test.go
+++ b/publicsuffix/example_test.go
@@ -7,6 +7,7 @@ package publicsuffix_test
 import (
 	"fmt"
 	"strings"
+	"testing"
 
 	"golang.org/x/net/publicsuffix"
 )
@@ -17,7 +18,8 @@ import (
 // managed, or unmanaged (not explicitly in the PSL).
 //
 // See https://publicsuffix.org/ for the underlying PSL data.
-func ExamplePublicSuffix_manager() {
+func ExamplePublicSuffix_manager(t *testing.T) {
+	t.SkipNow()
 	domains := []string{
 		"amazon.co.uk",
 		"books.amazon.co.uk",
diff --git a/publicsuffix/list_test.go b/publicsuffix/list_test.go
index 090c431..cf150ef 100644
--- a/publicsuffix/list_test.go
+++ b/publicsuffix/list_test.go
@@ -62,6 +62,7 @@ func TestFind(t *testing.T) {
 }
 
 func TestICANN(t *testing.T) {
+	t.SkipNow()
 	testCases := map[string]bool{
 		"foo.org":            true,
 		"foo.co.uk":          true,
@@ -295,6 +296,7 @@ func BenchmarkPublicSuffix(b *testing.B) {
 }
 
 func TestPublicSuffix(t *testing.T) {
+	t.SkipNow()
 	for _, tc := range publicSuffixTestCases {
 		gotPS, gotICANN := PublicSuffix(tc.domain)
 		if gotPS != tc.wantPS || gotICANN != tc.wantICANN {
@@ -304,6 +306,7 @@ func TestPublicSuffix(t *testing.T) {
 }
 
 func TestSlowPublicSuffix(t *testing.T) {
+	t.SkipNow()
 	for _, tc := range publicSuffixTestCases {
 		gotPS, gotICANN := slowPublicSuffix(tc.domain)
 		if gotPS != tc.wantPS || gotICANN != tc.wantICANN {
@@ -313,6 +316,7 @@ func TestSlowPublicSuffix(t *testing.T) {
 }
 
 func TestNumICANNRules(t *testing.T) {
+	t.SkipNow()
 	if numICANNRules <= 0 {
 		t.Fatal("no ICANN rules")
 	}
